prime-router/metadata/fhir_mapping/hl7/datatype/HumanName.yml

```diff
@@ -1,56 +1,95 @@
-    valueOf: 'GeneralUtils.generateName( prefix, first, middle, family, suffix)'
-    expressionType: JEXL
-    vars:
-      prefix: String, XPN.5 |XCN.6 |CNN.6 |NDL.6 | PPN.6
-      first: String, XPN.2 |XCN.3 |CNN.3 |NDL.3 | PPN.3
-      middle: String, XPN.3 |XCN.4 |CNN.4 |NDL.4 | PPN.4
-      family: String, XPN.1 |XCN.2 |CNN.2 |NDL.2 | PPN.2
-      suffix: String, XPN.4 |XCN.5 |CNN.5 |NDL.5 | PPN.5
+  type: STRING
+  valueOf: 'GeneralUtils.generateName(prefix, first, middle, family, suffix)'
+  expressionType: JEXL
+  vars:
+    prefix: String, XPN.5 | XCN.6 | CNN.6 | NDL.6 | PPN.6
+    first: String, XPN.2 | XCN.3 | CNN.3 | NDL.3 | PPN.3
+    middle: String, XPN.3 | XCN.4 | CNN.4 | NDL.4 | PPN.4
+    family: String, XPN.1 | XCN.2 | CNN.2 | NDL.2 | PPN.2
+    suffix: String, XPN.4 | XCN.5 | CNN.5 | NDL.5 | PPN.5
+
 family:
-     type: STRING
-     valueOf: XPN.1 |XCN.2 |CNN.2 |NDL.2 | PPN.2
-     expressionType: HL7Spec
+  type: STRING
+  valueOf: XPN.1 | XCN.2 | CNN.2 | NDL.2 | PPN.2
+  expressionType: HL7Spec
+
 given:
-    type: STRING
-    valueOf: 'GeneralUtils.makeStringArray( first, middle)'
-    generateList: true
-    expressionType: JEXL
-    vars:
-      first: String, XPN.2 | XCN.3 | CNN.3 | NDL.3 | PPN.3
-      middle: String, XPN.3 | XCN.4 | CNN.4 | NDL.4 | PPN.4
+  type: STRING
+  valueOf: 'GeneralUtils.makeStringArray(first, middle)'
+  generateList: true
+  expressionType: JEXL
+  vars:
+    first: String, XPN.2 | XCN.3 | CNN.3 | NDL.3 | PPN.3
+    middle: String, XPN.3 | XCN.4 | CNN.4 | NDL.4 | PPN.4
+
 prefix:
-     type: STRING
-     valueOf: XPN.5 | XCN.6 | CNN.6 |  NDL.6 | PPN.6
-     expressionType: HL7Spec
+  type: STRING
+  valueOf: XPN.5 | XCN.6 | CNN.6 |  NDL.6 | PPN.6
+  expressionType: HL7Spec
+
 use:
-    type: NAME_USE
-    valueOf: XPN.7 | XCN.10
-    expressionType: HL7Spec
-
-suffix: 
-     type: STRING
-     generateList: true
-     valueOf: XPN.4 | XCN.5 | CNN.5 | NDL.5 | PPN.5
-     expressionType: HL7Spec
-
-suffix_2:
-     type: STRING
-     generateList: true
-     valueOf: XCN.7
-     expressionType: HL7Spec
-
-suffix_3:
+  type: NAME_USE
+  valueOf: XPN.7 | XCN.10
+  expressionType: HL7Spec
+
+suffix_1-suffix:
+  type: STRING
+  generateList: true
+  valueOf: XPN.4 | XCN.5 | CNN.5 | NDL.5 | PPN.5
+  expressionType: HL7Spec
+
+suffix_2-degree:
+  type: STRING
+  generateList: true
+  valueOf: XCN.7
+  expressionType: HL7Spec
+
+suffix_3-professional-suffix:
   type: STRING
   generateList: true
   valueOf: XCN.21
   expressionType: HL7Spec
-     
+
 period:
-    valueOf: datatype/Period
-    expressionType: resource
-    vars:
-       start: XPN.12 | XCN.19
-       end: XPN.13 | XCN.20
-     
\ No newline at end of file
+  valueOf: datatype/Period
+  expressionType: resource
+  vars:
+    start: XPN.12 | XCN.19
+    end: XPN.13 | XCN.20
+
+extension:
+  generateList: true
+  expressionType: nested
+  condition: $suffix NOT_NULL || $suffixDegree NOT_NULL || $professionalSuffix NOT_NULL
```

Comment:
What do you think about have the conditions be applied at the expressions level rather than the top level?

Reply:
I love it, I thought this was required to prevent an empty `name` from appearing, but it seems like it's working actually, just committed the fix.